src/main/java/org/prebid/server/bidder/adquery/AdQueryBidder.java

```diff
@@ -0,0 +1,205 @@
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.adquery.ExtImpAdQuery;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.BidderUtil;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class AdQueryBidder implements Bidder<AdQueryRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpAdQuery>> AD_QUERY_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+    private static final String PREBID_VERSION = "server";
+    private static final String BIDDER_NAME = "adquery";
+    private static final String DEFAULT_CURRENCY = "PLN";
+    private static final String ORTB_VERSION = "2.5";
+    private static final String ADM_TEMPLATE = "<script src=\"%s\"></script>%s";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public AdQueryBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<AdQueryRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<HttpRequest<AdQueryRequest>> httpRequests = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            final ExtImpAdQuery extImpAdQuery;
+            try {
+                extImpAdQuery = parseImpExt(imp);
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+                continue;
+            }
+            httpRequests.add(createRequest(request, imp, extImpAdQuery));
+        }
+
+        return Result.of(httpRequests, errors);
+    }
+
+    private ExtImpAdQuery parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), AD_QUERY_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private HttpRequest<AdQueryRequest> createRequest(BidRequest bidRequest, Imp imp, ExtImpAdQuery extImpAdQuery) {
+        final AdQueryRequest outgoingRequest = createAdQueryRequest(bidRequest, imp, extImpAdQuery);
+
+        return HttpRequest.<AdQueryRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(resolveHeader(bidRequest.getDevice()))
+                .impIds(BidderUtil.impIds(bidRequest))
```

Comment:
since we build one request per imp it should be `.impIds(Collections.singleton(imp.getId))`